gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -420,5 +420,188 @@ public static ChangeRequestListOption sortOrder(SortingOrder order) {
+   * restricted by specifying {@link ProjectOption}s.
+   *
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/projects/get">Cloud DNS Projects: get</a>
+   */
+  ProjectInfo getProjectInfo(ProjectOption... fields);
+
+  /**
+   * Returns the current project id.
+   */
+  String getProjectId();
+
+  /**
+   * Returns the current project number.
+   */
+  BigInteger getProjectNumber();
+
+  /**
+   * Submits a change request for the specified zone. The returned object contains the following
+   * read-only fields supplied by the server: id, start time and status. time, id, and list of name
+   * servers. The fields to be returned can be selected by {@link ChangeRequestOption}s.
+   *
+   * @return the new {@link ChangeRequest} or {@code null} if zone is not found
+   * @throws DnsException upon failure
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/create">Cloud DNS Changes: create</a>
+   */
+  ChangeRequest applyChangeRequest(BigInteger zoneId, ChangeRequest changeRequest,
+      ChangeRequestOption... options);
+
+  /**
+   * Submits a change request for the specified zone. The returned object contains the following
+   * read-only fields supplied by the server: id, start time and status. time, id, and list of name
+   * servers. The fields to be returned can be selected by {@link ChangeRequestOption}s.
+   *
+   * @return the new {@link ChangeRequest}
+   * @throws DnsException upon failure if zone is not found
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/create">Cloud DNS Changes: create</a>
+   */
+  ChangeRequest applyChangeRequest(String zoneName, ChangeRequest changeRequest,
+      ChangeRequestOption... options);
+
+  /**
+   * Retrieves updated information about a change request previously submitted for a zone identified
+   * by ID. Returns {@code null} if the request cannot be found and throws an exception if the zone
+   * does not exist. The fields to be returned using can be specified using {@link
+   * ChangeRequestOption}s.
+   *
+   * @throws DnsException upon failure or if the zone cannot be found
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/get">Cloud DNS Chages: get</a>
+   */
+  ChangeRequest getChangeRequest(String changeRequestId, BigInteger zoneId,
+      ChangeRequestOption... options);
+
+  /**
+   * Retrieves updated information about a change request previously submitted for a zone identified
+   * by ID. Returns {@code null} if the request cannot be found and throws an exception if the zone
+   * does not exist. The fields to be returned using can be specified using {@link
+   * ChangeRequestOption}s.
+   *
+   * @throws DnsException upon failure or if the zone cannot be found
+   * @see <a href="https://cloud.google.com/dns/api/v1/changes/get">Cloud DNS Chages: get</a>
+   */
+  ChangeRequest getChangeRequest(String changeRequestId, String zoneName,
```

Comment:
To me me the order generic to specific (zoneName, changeRequestId) is more natural and more consistent with apply.